tests/phpunit/tests/user.php

```diff
@@ -605,43 +605,215 @@ public function test_count_many_users_posts() {
 	}
 
 	/**
-	 * Tests salted cache functionality for count_many_users_posts().
+	 * Ensure the second and subsequent calls to count_many_users_posts() are cached.
 	 *
-	 * @ticket 63405
+	 * @ticket 63045
 	 */
-	public function test_count_many_users_posts_salted_cache() {
+	public function test_count_many_users_posts_is_cached() {
+		$user_1 = self::$user_ids[0];
+		$user_2 = self::$user_ids[1];
 
-		$author_id = self::factory()->user->create( array( 'role' => 'author' ) );
-		$editor_id = self::factory()->user->create( array( 'role' => 'editor' ) );
+		// Create posts for both users.
+		self::factory()->post->create( array( 'post_author' => $user_1 ) );
+		self::factory()->post->create( array( 'post_author' => $user_2 ) );
 
-		// Create posts for both users
-		self::factory()->post->create( array( 'post_author' => $author_id ) );
-		self::factory()->post->create( array( 'post_author' => $editor_id ) );
+		// Warm the cache.
+		count_many_users_posts( array( $user_1, $user_2 ), 'post', false );
```

Comment:
Minor point: It would make sense to me to obtain the return value of this and add `assertSame()` for the second call below, as a sanity check.